File: AD-HOC/C#/TELA_3/1383.cs
using System;

class Program
{
    static string IsSudoku(int[,] matriz)
    {
        for (int i = 0; i < 9; i++)
        {
            int[] testador = new int[10];
            for (int j = 0; j < 9; j++)
            {
                if (testador[matriz[i, j]] != 0)
                {
                    return "NAO";
                }
                else
                {
                    testador[matriz[i, j]] = 1;
                }
            }
        }

        for (int i = 0; i < 9; i++)
        {
            int[] testador = new int[10];
            for (int j = 0; j < 9; j++)
            {
                if (testador[matriz[j, i]] != 0)
                {
                    return "NAO";
                }
                else
                {
                    testador[matriz[j, i]] = 1;
                }
            }
        }

        for (int i = 2; i < 9; i += 3)
        {
            int[] testador = new int[10];
            for (int j = i - 2; j <= i; j++)
            {
                for (int k = i - 2; k <= i; k++)
                {
                    if (testador[matriz[j, k]] != 0)
                    {
                        return "NAO";
                    }
                    else
                    {
                        testador[matriz[j, k]] = 1;
                    }
                }
            }
        }

        return "SIM";
    }

    static void Main()
    {
        int n = int.Parse(Console.ReadLine());

        for (int h = 1; h <= n; h++)
        {
            int teste = 0;
            int[,] matriz = new int[9, 9];

            for (int i = 0; i < 9; i++)
            {
                string[] input = Console.ReadLine().Split();
                for (int j = 0; j < 9; j++)
                {
                    matriz[i, j] = int.Parse(input[j]);
                }
            }

            string saida = IsSudoku(matriz);

            Console.WriteLine($"Instancia {h}");
            Console.WriteLine(saida);
            Console.WriteLine();
        }
    }
}

File: AD-HOC/PYTHON/TELA_3/1383.py
def is_sudoku(matriz):
    for i in range(9):
        testador = [0] * 10
        for j in range(9):
            if testador[matriz[i][j]] != 0:
                return 'NAO'
            else:
                testador[matriz[i][j]] = 1

    for i in range(9):
        testador = [0] * 10
        for j in range(9):
            if testador[matriz[j][i]] != 0:
                return 'NAO'
            else:
                testador[matriz[j][i]] = 1

    for i in range(2, 9, 3):
        testador = [0] * 10
        for j in range(i - 2, i + 1):
            for k in range(i - 2, i + 1):
                if testador[matriz[j][k]] != 0:
                    return 'NAO'
                else:
                    testador[matriz[j][k]] = 1

    return 'SIM'


n = int(input())
for h in range(1, n + 1):
    teste = 0
    matriz = [list(map(int, input().split())) for _ in range(9)]

    for i in range(9):
        testador = [0] * 10
        for j in range(9):
            if testador[matriz[i][j]] != 0:
                teste = 1
            else:
                testador[matriz[i][j]] = 1

    for i in range(9):
        testador = [0] * 10
        for j in range(9):
            if testador[matriz[j][i]] != 0:
                teste = 1
            else:
                testador[matriz[j][i]] = 1

    for i in range(2, 9, 3):
        testador = [0] * 10
        for j in range(i - 2, i + 1):
            for k in range(i - 2, i + 1):
                if testador[matriz[j][k]] != 0:
                    teste = 1
                else:
                    testador[matriz[j][k]] = 1

    print(f"Instancia {h}")
    print("SIM" if not teste else "NAO")
    print()
